test/e2e/test_cluster_configs/flatcar/flatcar.json

```diff
@@ -0,0 +1,59 @@
+{
+	"apiModel": {
+		"apiVersion": "vlabs",
+		"properties": {
+			"orchestratorProfile": {
+				"orchestratorType": "Kubernetes",
+				"kubernetesConfig": {
+					"addons": [
+						{
+							"name": "cluster-autoscaler",
+							"enabled": true,
+							"pools": [
+								{
+									"name": "pool1",
+									"config": {
+										"min-nodes": "3",
+										"max-nodes": "9"
+									}
+								}
+							],
+							"config": {
+								"scan-interval": "1m",
+								"scale-down-delay-after-add": "1m0s",
+								"scale-down-unneeded-time": "2m0s"
+							}
+						}
+					]
+				}
+			},
+			"masterProfile": {
+				"count": 1,
+				"dnsPrefix": "",
+				"vmSize": "Standard_D2_v3"
+			},
+			"agentPoolProfiles": [

```

Comment:
@jackfrancis Just reviewed these + the options defined in https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#agentPoolProfiles

LGTM for `agentPoolProfiles` test configs.